examples/opt_serving/model/wrapper.py

```diff
@@ -219,14 +250,22 @@ def get_hf_opt_model(model_name, device, num_beams):
 
     def inference_func(input_ids,
                        past_key_values,
+                       attention_mask,
                        output_attentions=False,
                        output_hidden_states=False):
+        attention_length = attention_mask.size()[-1]
         if past_key_values is None:
-            attention_mask = None
+            past_length = 0
         else:
             past_length = past_key_values[0][0].shape[2]
-            attention_mask = torch.ones(
-                (input_ids.shape[0], past_length + 1)).to(device)
+        if past_length >= attention_length:
+            attention_mask = torch.cat(
+                (attention_mask,
+                 torch.ones((input_ids.shape[0],
+                             past_length + 1 - attention_length)).to(device)),
+                dim=1)
+        elif attention_mask != None:
```

Comment:
`attention_mask` will never be `None` here since you take the size of it on line 256.

Reply:
@ZYHowell